### PR TITLE
Refactored provider and use higher order compoents

### LINKED ISSUE #5: Refactor Request provider
Right now, JS classes are used to generate requests (providers in `util`).
We could save a lot of code and make it more maintainable.

### PR DESCRIPTION
**Pull request by jgraeger**
Fixes #5

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -8,26 +8,21 @@ import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
...
-        setFlow(event.target.value);
-    }
+    const requests = requestProvider(cluster, flow);
 
     const checkSession = async () => {
-        let login = new LoginProvider(cluster, flow);
-        var jsonRes = await API.makeRequest(login.getSessionRequest());
-
+        const jsonRes = await API.makeRequest(requests.session());
         const isSessionActive = !(jsonRes.status == 401);
```
Comment (ldb) [inline src/App.jsx]:
```suggestion
        const isSessionActive = !(jsonRes.status === 401);
```

if there is anything I remember from my time as a nodeJS dev it's that you should always check type equality :D

File: src/pages/ApiPage.jsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            </Row>
+        </Container>
+    );
+};
+
+export const defaultInitReducer = ({ request, response }) => {
+    try {
+        const flowID = response.json.id;
+        const csrf = response.json.ui.nodes
+            .filter(x => x.attributes.name == "csrf_token")
```
Comment (ldb) [inline src/pages/ApiPage.jsx:101]: Export as const?

**Review by ldb — COMMENTED** [review]
Take with a handful of grains of salt :D

File: src/pages/ApiPage.jsx (hunk 2 above)
Comment (jgraeger) [reply src/pages/ApiPage.jsx:101]: Yeah. Thats by intention and actually used. A reducer returns a plain object. So if you want to not override the default reducer, but instead extend it's output, you can mix the exported default reducers in. Example:

```js
import { defaultInitReducer } from 'ApiPage';
// [...]

initializer: {
    fn: someRequestGenerator,
    reducer: ({request, reponse}) => ({ 
        ...defaultInitReducer({request, response}),
        foo: response.bar
    }),
};
```

File: src/App.jsx (hunk 1 above)
Comment (jgraeger) [reply src/App.jsx]: Indeed. Happens in a language that is basically unusable without a linter :D 
I created #7 as a consequence.

File: src/pages/ApiPage.jsx (hunk 2 above)
Comment (jgraeger) [reply src/pages/ApiPage.jsx:101]: Here, the `defaultSubmitReducer` is used by the RegPage for that purpose.
[Line 23](https://github.com/mindtastic/stagefright/blob/e778a853cc41fadb9f2c6f9ba6636d8e83d2ad09/src/pages/RegPage.jsx).

Comment (ldb) [reply src/pages/ApiPage.jsx:101]: I mean just the csrf_token string

**Review by theEpsilon — APPROVED** [review]
So this is what "fluent in JavaScript" actually means :D

much impressed

**Pull request merged**